oas.go

```diff
@@ -295,3 +295,8 @@ type TagDefinition struct {
 	Name        string          `json:"name"`
 	Description *ReffableString `json:"description,omitempty"`
 }
+
+type MapDefinition struct {
```

Comment:
> We can't just have a string for MapDefinition because when it's marshalled it considers nil an empty string.

When does this struct get marshalled? It appears to only be used in the `parsePackageAliases` functions:

```
func (p *parser) parsePackageAliases() error {
                                  ...
				case "@packagealias":
					m, err := parsePackageAliases(comment)

					if err != nil {
						return err
					}
					p.PackageAliases[m.Name] = m.NewName
				}
```

```
func parsePackageAliases(comment string) (*MapDefinition, error) {
	re, _ := regexp.Compile("\"([^\"]*)\"")
	matches := re.FindAllStringSubmatch(comment, -1)
	...
	renameMap := MapDefinition{Name: matches[0][1], NewName: &matches[1][1]}

	return &renameMap, nil
}
```

My suggestion was to just have this function return the two strings instead of wrapping them in a struct,

```
func (p *parser) parsePackageAliases() error {
                                  ...
				case "@packagealias":
					name, alias, err := parsePackageAliases(comment)

					if err != nil {
						return err
					}
					p.PackageAliases[name] = alias
				}
```

```
func parsePackageAliases(comment string) (string, string, error) {
	re, _ := regexp.Compile("\"([^\"]*)\"")
	matches := re.FindAllStringSubmatch(comment, -1)
	...
	return matches[0][1], matches[1][1], nil
}
```

Reply:
you're entirely correct. It's updated and much cleaner now, also cleaned up associated tests.